docs/developer/best-practices/stability.asciidoc

```diff
@@ -0,0 +1,67 @@
+[[stability]]
+=== Stability
+
+Ensure your feature will work under all possible {kib} scenarios.
+
+[float]
+==== Environmental configuration scenarios
+
+* Cloud
```

Comment:
This is probably fine to include here for simplicity, but many of the cloud environment questions aren't relevant to the community.

Reply:
Good point. Maybe we should extract out what exactly that means. like instead of "does it run on cloud", say, "Will this work with servers that have port forwarding turned on".  But I am not confident enough of the specifics to make those changes right now. Noted for the future.